src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -1,59 +1,92 @@
 import * as React from 'react';
+import { MdClose } from 'react-icons/md';
+import useResizeObserver from 'use-resize-observer';
 import { bem } from '../../../utils';
-import { Heading } from '../../Heading';
 import { Button } from '../../Buttons';
 import { Modal, ModalProps } from '../../Modal';
+import { Heading } from '../../Heading';
+import { ButtonContext } from '../../../constants/component-specific';
 import styles from './Dialog.scss';
 
 export type ButtonProps = {
     onClick: () => void;
     label: string;
 };
 
-export interface DialogProps extends ModalProps {
+export type HeaderProps = {
+    /** Dialog header title */
+    title: string | undefined;
+    /** Properties of the close button */
+    closeButton: ButtonProps;
```

Comment:
We also have a prop from Modal Props called `onRequestClose`. Maybe we should omit it from the dialog, and instead pass this closeButton.onClick to it? What do you think?